engine/src/main/java/io/camunda/zeebe/engine/state/message/DbProcessMessageSubscriptionState.java

```diff
@@ -64,6 +64,14 @@ public DbProcessMessageSubscriptionState(
             transactionContext,
             sentTimeCompositeKey,
             DbNil.INSTANCE);
+
+    subscriptionColumnFamily.whileTrue(
+        (compositeKey, subscription) -> {
+          if (subscription.isOpening() || subscription.isClosing()) {
+            transientState.add(subscription.getRecord());
+          }
+          return true;
+        });
```

Comment:
Let's discuss implementation options:
* One option would be to implement a separate listener. Then in order for this listener to work I would have to expose the `add()` method in the interface of the transient/pending state. I am reluctant to do this, because it would break the encapsulation I had in mind, whereby the persistent state is the SSOT regarding which items are in the transient/pending state, and the transient/pending interface only allows to change the sent time on those items. If I expose an `add()` method than I loose the SSOT, but it would also help with the weird lines in the migration code, where I call the persistent state for its side effects on the transient state
* Another option would be to register `DbProcessMessageSubscriptionState` directly as listener. This would allow privileged access to the transient state, and I think it is a nice addition if these state classes can hook into lifecycle events. 

What do you think?

Reply:
The second option sounds good to me.